### PR TITLE
Feature/ele 285 restructurerefactor navigation wrapper into view wrapper v2

### PR DESCRIPTION
**Pull request by dannelundqvist**
Restructured navigation code, streamlined navigation and history state initialization/handling. Added ViewWrapper, ViewProvider and hook for handling active view better, removed rerendering of content.

### CODE REVIEW

File: src/navigation/lib/initializeNavigationState.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+          </ViewWrapper>
+        </ViewProvider>
+      )]
+    }
+  }
+
+
+  const preContent = history.state.contentState.map((item: ContentState): { name: string } => {
+    return item
+  })
```
Comment (gularsson) [inline src/navigation/lib/initializeNavigationState.tsx:53]:
```suggestion
  // Recreate navigationstate from history
  const preContent = history.state.contentState.map((item: ContentState): { name: string } => {
    return item
  })
```
Just for clarity, your choice!

File: src/navigation/lib/navigationReducer.tsx [hunk 2]
```diff
@@ -15,7 +18,8 @@ export function navigationReducer(state: NavigationState, action: NavigationActi
         throw new Error('Content is undefined')
       }
 
-      const views = calculateViews(state, action.content)
+      // const views = calculateViews(state, screenDefinitions, action.content)
```
Comment (gularsson) [inline src/navigation/lib/navigationReducer.tsx:21]: :broom:

**Review by gularsson — CHANGES_REQUESTED** [review]
Works great when testing, we're rid of some of the bugs that we previously had. Seems like a good restructuring, less code, better namning and division.
I'm having a little bit of trouble grasping everything, but that will come with time. My comments are therefor mostly of a petty character

File: src/navigation/lib/calculateViews.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+} from '@/lib/getScreenDefinitions'
+
+/**
+ * Calculate each views colSpan that is in the current content
+ *
+ * @param viewRegistry ViewRegistry
+ * @param content ContentState[]
+ * @returns Array<{ name: string, colSpan: number }>
+ */
+export function calculateViewWidts(
```
[the remark this answers is not in the document]
Comment (dannelundqvist) [reply src/navigation/lib/calculateViews.ts]: Renaming file as well to follow function name. Fixed imports to follow convention.

File: src/navigation/lib/initializeNavigationState.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+export function initializeNavigationState(): NavigationState {
+  Object.keys(views).forEach((name) => {
+    registeredComponents.set(name, {
+      component: views[name as View],
+      meta: views[name as View].meta
+    })
+  })
+
+  const { name = 'start', props } = currentView()
```
[the remark this answers is not in the document]
Comment (dannelundqvist) [reply src/navigation/lib/initializeNavigationState.tsx]: Correct

File: src/navigation/lib/initializeNavigationState.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  if (!history?.state?.contentState?.length) {
+    const InititalView = viewRegistry.getByPath(window.location.pathname)
+    const id = uuid.v4()
```
[the remark this answers is not in the document]
Comment (dannelundqvist) [reply src/navigation/lib/initializeNavigationState.tsx]: Agree that it is good with comments here. Also fixed imports to follow convention in the same fix.

**Comment by dannelundqvist** [comment]
Not petty. The code is now better for it. I found more instances of not following the import conventions. Fixed those as well.

**Review by gularsson — APPROVED** [review]

**Pull request merged**